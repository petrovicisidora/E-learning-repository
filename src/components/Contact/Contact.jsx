//Contact component with contact information

import React from "react";

import {
  StyledContactContainer,
  StyledContactEmail,
  StyledContactName,
  StyledContactTitle,
} from "./Contact.style";

function Contact({ title, name, email }) {
  return (
    <StyledContactContainer>
      <StyledContactTitle>{title}</StyledContactTitle>
      <StyledContactName>{name}</StyledContactName>
      <StyledContactEmail>
        <a href={`mailto:${email}`}>{email}</a>
      </StyledContactEmail>
    </StyledContactContainer>
  );
}

export default Contact;
