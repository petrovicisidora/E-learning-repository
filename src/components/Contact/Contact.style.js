import styled from "styled-components";

export const StyledContactContainer = styled.div`
  width: 100%;
  padding-left: 20px;
  margin-bottom: 30px;
`;

export const StyledContactTitle = styled.h2`
  font-weight: bold;
  font-size: 14px;
  color: gray;
  margin: 5px 0;
`;

export const StyledContactName = styled.p`
  font-size: 12px;
  color: gray;
  margin: 5px 0;
`;

export const StyledContactEmail = styled.p`
  font-size: 12px;
  margin: 5px 0;
`;
