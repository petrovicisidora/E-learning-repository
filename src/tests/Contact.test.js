import React from "react";
import { render, screen } from "@testing-library/react";

import Contact from "../components/Contact/Contact";

const contactProps = {
  title: "Test Title",
  name: "Test Name",
  email: "test@email.com",
};

test("renders contact's elements correctly", () => {
  render(<Contact {...contactProps} />);
  const contactTitle = screen.getByText("Test Title");
  const contactName = screen.getByText("Test Name");
  const contactEmail = screen.getByText("test@email.com");

  expect(contactTitle).toBeInTheDocument();
  expect(contactName).toBeInTheDocument();
  expect(contactEmail).toBeInTheDocument();
});

test("contact has correct styled", () => {
  render(<Contact {...contactProps} />);
  const contactTitle = screen.getByText("Test Title");
  const contactName = screen.getByText("Test Name");

  expect(contactTitle).toHaveStyle("font-size: 14px");
  expect(contactTitle).toHaveStyle("color: gray");
  expect(contactName).toHaveStyle("margin: 5px 0");
});

test("matches snapshot", () => {
  const { container } = render(<Contact />);
  expect(container).toMatchSnapshot();
});
