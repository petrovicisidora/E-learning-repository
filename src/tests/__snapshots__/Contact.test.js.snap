// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-beySPh ijhkct"
  >
    <h2
      class="sc-guDLey gFpTAT"
    />
    <p
      class="sc-dmyCSP juzinH"
    />
    <p
      class="sc-hLQSwg gRdAjK"
    >
      <a
        href="mailto:undefined"
      />
    </p>
  </div>
</div>
`;
